src/main/java/com/google/devtools/build/skydoc/renderer/RendererMain.java

```diff
@@ -184,5 +222,50 @@ private static void printAspectInfos(
     }
   }
 
+  private static void printRepositoryRuleInfos(
+      PrintWriter printWriter, MarkdownRenderer renderer, List<RepositoryRuleInfo> ruleInfos)
+      throws IOException {
+    // Repository rules are printed sorted by their qualified name, and their attributes are sorted
+    // by name, with ATTRIBUTE_ORDERING specifying a fixed sort order for some standard attributes.
+    ImmutableList<RepositoryRuleInfo> sortedRepositoryRuleInfos =
+        ruleInfos.stream()
+            .map(RendererMain::withSortedRuleAttributes)
+            .sorted(comparing(RepositoryRuleInfo::getRuleName))
+            .collect(toImmutableList());
+    for (RepositoryRuleInfo repositoryRuleProto : sortedRepositoryRuleInfos) {
+      printWriter.println(renderer.render(repositoryRuleProto.getRuleName(), repositoryRuleProto));
+      printWriter.println();
+    }
+  }
+
+  private static void printModuleExtensionInfos(
+      PrintWriter printWriter, MarkdownRenderer renderer, List<ModuleExtensionInfo> ruleInfos)
+      throws IOException {
+    // Module extension are printed sorted by their qualified name, and their tag classes'
+    // attributes are sorted by name, with ATTRIBUTE_ORDERING specifying a fixed sort order for some
+    // standard attributes.
+    ImmutableList<ModuleExtensionInfo> sortedModuleExtensionInfos =
+        ruleInfos.stream()
+            .map(RendererMain::withSortedTagAttributes)
+            .sorted(comparing(ModuleExtensionInfo::getExtensionName))
+            .collect(toImmutableList());
+    for (ModuleExtensionInfo moduleExtensionProto : sortedModuleExtensionInfos) {
+      printWriter.println(
+          renderer.render(moduleExtensionProto.getExtensionName(), moduleExtensionProto));
+      printWriter.println();
+    }
+  }
+
+  /**
+   * Formats the label of the .bzl module being documented for use in documentation, falling back to
+   * "..." if the label is not available.
+   */
+  // TODO(arostovtsev): we ought to add an option to semi-canonicalize .bzl file labels in the local
```

Comment:
I can look into fixing this as a follow-up based on `native.module_name`.

Reply:
Unfortunately, `native.module_name` can only be used in the loading phase :/

Reply:
My idea was to add a `module_name` parameter to the `stardoc` macro that uses `native.module_name` as the default.

Reply:
On further reflection, I think the right place to fix this should be in `starlark_doc_extract` - we want to be able to optionally provide the local repo name for all labels (including e.g. label attribute defaults), in the proto output as well as in markdown.

The `starlark` macro can then provide the local repo map (obtained from `native.module_name` or a manual override for cases where the recommended repo name for a module's users is different from the module name) to the `starlark_doc_extract` target.

Reply:
When you say "local repo name", which name are you referring to exactly? The apparent name used by the root module? 

What would comprise the "local repo map" that you would want to have passed into the rule?

Reply:
By "local repo", I mean the repo of the `stardoc` (and `starlark_doc_extract`) target.

Currently, `starlark_doc_extract` treats that repo as the main repo for the purpose of stringifying labels in proto output. This behavior made sense to me at the time when I implemented it because it meant `starlark_doc_extract`'s output is the same regardless of whether doc targets are being built in their own repo or by a user of the repo, and thus made golden tests sane/predictable.

In retrospect, the correct behavior should have been to allow the name of the local repo to be remapped to a custom one when stringifying labels.

Reply:
I submitted https://github.com/bazelbuild/bazel/pull/19002, please let me know whether this is what you had in mind.